mergify_engine/gitter.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright © 2021 Mergify SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import asyncio
+import dataclasses
+import logging
+import shutil
+import tempfile
+import typing
+import urllib.parse
+
+from mergify_engine import config
+
+
+@dataclasses.dataclass
+class GitError(Exception):
+    returncode: int
+    output: str
+
+
+@dataclasses.dataclass
+class Gitter(object):
+    logger: logging.LoggerAdapter
+    tmp: str = dataclasses.field(init=False)
+
+    # Worker timeout at 5 minutes, so ensure subprocess return before
+    GIT_COMMAND_TIMEOUT: int = dataclasses.field(init=False, default=4 * 60 + 30)
```

Comment:
can we use the 5 minutes your referring to in the `default`? I guess it comes from a variable.

Reply:
I see

Reply:
I don't find a trace about that this comment mean, I guess it comes from the time we having celery.

Reply:
ok we might need to fix that then…